src/find_ice_labels.jl

```diff
@@ -84,3 +84,172 @@ function find_ice_labels(
+    _, heights = imhist(arr, imgtype)
+
+    locs, heights, _ = Peaks.findmaxima(heights)
+
+    # TODO: make this conditional on input args
+    order = sortperm(heights; rev=true)
+    locs, heights = locs[order], heights[order]
+
+    return (locs=locs, heights=heights)
+end
+
+
+function get_ice_labels_mask(ref_img::Matrix{RGB{N0f8}}, thresholds, factor=1)
+    cv = channelview(ref_img)
+    cv = [float64.(cv[i, :, :]) .* factor for i in 1:3]
+    mask_ice_band_7 = cv[1] .< thresholds[1]
+    mask_ice_band_2 = cv[2] .> thresholds[2]
+    mask_ice_band_1 = cv[3] .> thresholds[3]
+    mask = mask_ice_band_7 .* mask_ice_band_2 .* mask_ice_band_1
+    @debug "Found $(sum(mask)) ice pixels"
+    return mask
+end
+
+# TODO: add test
+function get_nlabel(
+    ref_img,
+    morph_residue_labels,
+    factor;
+    band_7_threshold::T=5,
+    band_2_threshold::T=230,
+    band_1_threshold::T=240,
+    band_7_threshold_relaxed::T=10,
+    band_1_threshold_relaxed::T=190,
+    possible_ice_threshold::T=75,
+) where {T<:Integer}
+    _getnlabel(morphr, mask) = StatsBase.mode(morphr[mask])
+
+    # Initial attempt to get ice labels
+    thresholds = (band_7_threshold, band_2_threshold, band_1_threshold)
+    ice_labels_mask = get_ice_labels_mask(ref_img, thresholds, 255)
+    sum(ice_labels_mask) > 1 && return _getnlabel(morph_residue_labels, ice_labels_mask)
+
+    # First relaxation
+    thresholds = (band_7_threshold_relaxed, band_2_threshold, band_1_threshold_relaxed)
+    ice_labels_mask = get_ice_labels_mask(ref_img, thresholds, 255)
+    sum(ice_labels_mask) > 0 && return _getnlabel(morph_residue_labels, ice_labels_mask)
+
+    # Second/Third relaxation
+    return get_nlabel_relaxation(
+        ref_img,
+        morph_residue_labels,
+        factor,
+        possible_ice_threshold,
+        band_7_threshold_relaxed,
+        band_2_threshold,
+    )
+end
+
+function get_nlabel_relaxation(
+    ref_img,
+    morph_residue_labels,
+    factor,
+    possible_ice_threshold,
+    band_7_threshold_relaxed,
+    band_2_threshold,
+)
+    # filter b/c channels (landmasked channels 2 and 3) and compute peaks
+    b, c = [float64.(channelview(ref_img)[i, :, :]) .* factor for i in 2:3]
+    b[b .< possible_ice_threshold] .= 0
+    c[c .< possible_ice_threshold] .= 0
+    pksb, pksc = get_image_peaks.([b, c])
+
+    # return early if no peaks are found
+    !all(length.([pksb.locs, pksc.locs]) .> 2) && return 1
+
+    relaxed_thresholds = [band_7_threshold_relaxed, pksb.locs[2], pksc.locs[2]]
+    ice_labels = get_ice_labels_mask(ref_img, relaxed_thresholds, factor)
+
+    sum(ice_labels) > 0 && return StatsBase.mode(morph_residue_labels[ice_labels])
+
+    # Final relaxation
+    mask_b = b .> band_2_threshold
+    sum(mask_b) > 0 && return StatsBase.mode(morph_residue_labels[mask_b])
+
+    # TODO: Should a fallback value be added? Return nothing if no ice is found? return 1? throw error?
+end
+
+# TODO: Add tests
```

Comment:
delete line 176, it seems like `test/test-tiled-ice-labels.jl` covers this script?